src/Doi.php

```diff
@@ -3,10 +3,39 @@
 
 class Doi
 {
+    const PATTERN = <<<'EOT'
+{
+    10 # Directory indicator (always 10)
+    \.
+    (?:
+        # ISBN-A
+        97[89]\. # ISBN (GS1) Bookland prefix
+        \d{2,8}  # ISBN registration group element and publisher prefix
+        /        # Prefix/suffix divider
+        \d{1,7}  # ISBN title enumerator and check digit
+        |
+        # DOI
+        \d{4,9} # Registrant code
+        /       # Prefix/suffix divider
+        \S+     # DOI suffix
+    )
+}x
+EOT;
+    const EARLY_WILEY_SUFFIX = '/2-#$/';
+
     public static function extract($str)
     {
-        preg_match_all('#\b10\.(?:97[89]\.\d{2,8}/\d{1,7}|\d{4,9}/\S+)\b#', $str, $matches);
+        preg_match_all(self::PATTERN, $str, $matches);
+
+        return array_map([__CLASS__, 'stripTrailingPunctuation'], $matches[0]);
+    }
+
+    private static function stripTrailingPunctuation($doi)
+    {
+        if (!preg_match('/[[:punct:]]$/', $doi) || preg_match('/\(.+\)$/', $doi) || preg_match(self::EARLY_WILEY_SUFFIX, $doi)) {
+            return strtolower($doi);
```

Comment:
As we're using a method named `stripTrailingPunctuation` here, how would you feel about moving `strtolower()` out of the method? It feels like calling it within each `return` is indicating that it's actually a separate task from the punctuation stripping.